File: contains-duplicate.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

bool containsDuplicate(vector<int>& nums) {
  map<int, bool> nums_map;
  for (int num: nums) {
    if (nums_map[num] == true)
      return true;
    nums_map[num] = true;
  }
  return false;
}

int main(int argc, char const *argv[]) {
  vector<int> input = {1,1,1,3,3,4,3,2,4,2};
  cout << containsDuplicate(input);

  return 0;
}

File: contains-duplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        if len(nums) <= 1:
            return False
        my_set = set()
        for num in nums:
            if num in my_set:
                return True
            else:
                my_set.add(num)

        return False


s = Solution()
result = s.containsDuplicate([1, 2, 3, 1])
print(result)
